pandas/tests/frame/test_indexing.py

```diff
@@ -2590,3 +2590,24 @@ def test_head_tail(self):
         empty_df = DataFrame()
         assert_frame_equal(empty_df.tail(), empty_df)
         assert_frame_equal(empty_df.head(), empty_df)
+
+    def test_type_error_multiindex(self):
+        # See gh-12218
+        invalid_key = "is an invalid key"
+        not_raised = "TypeError not raised."
+
+        df = pd.DataFrame(columns=['i', 'c', 'x', 'y'],
+                          data=[[0, 0, 1, 2], [1, 0, 3, 4],
+                                [0, 1, 1, 2], [1, 1, 3, 4]])
+        dg = df.pivot_table(index='i', columns='c',
+                            values=['x', 'y'])
+
+        with assertRaisesRegexp(TypeError, invalid_key):
+            str(dg[:, 0])
+
+        # Neither of the following commands should
+        # throw a TypeError because they are valid
```

Comment:
you don't need to put these asserts around here, just run the actual commands. if they raise the tests will fail; this just obfuscuates tings.  best is actually to test the results

e.g.

```
result = dg.loc[:, (slice(None), 0)]
expected = ....
assert_.....
```